s3deploy.js

```diff
@@ -2,212 +2,192 @@ const { info, error, logWithSpinner, stopSpinner } = require('@vue/cli-shared-ut
 const path = require('path')
 const fs = require('fs')
 const mime = require('mime-types')
+const globby = require('globby')
 const AWS = require('aws-sdk')
 const PromisePool = require('es6-promise-pool')
 
-module.exports = async (options, api) => {
-  info(`Options: ${JSON.stringify(options)}`)
+const S3 = new AWS.S3()
 
-  AWS.config.update({
-    region: options.region,
-    httpOptions: {
-      connectTimeout: 10 * 1000,
-      timeout: 10 * 1000
-    }
-  })
+function contentTypeFor (filename) {
+  return mime.lookup(filename) || 'application/octet-stream'
+}
+
+async function bucketExists (options) {
```

Comment:
This function is pretty big and does more than just check if the bucketExists.
It can probably be broken up into 2-3 more focused functions.

Reply:
Ah, good call! Would make testing easier too :)